### ISSUE TITLE
Port of EasyMotion plugin

### ISSUE THREAD

**Issue by loonies** [vrapper/vrapper#134]
Any chance we get EasyMotion plugin ported?

http://www.vim.org/scripts/script.php?script_id=3526

**Comment by eulerreich** [comment]
+1 for this as well

**Comment by ilazarte** [comment]
+1

**Comment by mmbradle** [comment]
+1

**Comment by zhangll** [comment]
+1

**Comment by mmbradle** [comment]
I've been using vim-sneak lately and have thought it even better than easymotion. It may be easier to implement as well as it only requires highlighting of matches instead of the a,b,c,d... thing that easymotion does.

**Comment by albertdev** [comment]
@mmbradle 
I have to agree on that, simple highlighting would be a lot easier than drawing all those little overlays. It would still be complicated due to the way that vim-sneak integrates with Vim, but maybe it's better do-able.

Maybe create a new feature request with how you see this in Vrapper?

**Comment by loonies** [comment]
Agreed. Another ticket as feature request for port of _vim-sneak_ plugin.

I didn't try _vim-sneak_ yet, hopefully it's a good trade off.

**Comment by hrj** [comment]
:+1: for easy motion with overlays.

I just discovered easy motion and immediately came to check if vrapper supports it. I am now torn between two choices:
* Use vim + easy motion (or spacemacs) and avoid using Eclipse entirely
* Avoid easy motion

**Comment by NoAnyLove** [comment]
+1
Super useful plugin. Vim-sneak port is also good alternative choice.

**Comment by JakubNer** [comment]
+1, ty for vrapper

**Comment by rkhitin** [comment]
+1!

**Comment by albertdev** [comment]
For those of you who already want a zippier way to navigate, we're currently hosting a port of vim-sneak in our unstable repository.

It's not final yet but more of a sneak preview (pun oh so much intended). All comments are welcome on the pull request at #764, though we're aware that mostly the out-of-box key mapping is open to improvement.

**Comment by skybber** [comment]
There is a new and extremly promising plugin [EclipseAceJump
](https://github.com/tobiasmelcher/EclipseAceJump), may be the vrapper plugin can be base on this !

**Comment by 9527KGG** [comment]
+1, 
5 years after initial request ^_^

**Comment by DXist** [comment]
@9527KGG , you may look into https://youtrack.jetbrains.com/issue/VIM-820 as well. Probably,  JetBrains might be faster)

**Comment by albertdev** [comment]
@DXist I'm not sure if you're being serious or sarcastic.

(Fry meme goes here)

Anyway, since I've just let a few old prototypes loose on the world I might as well pick this one up again.

**Comment by albertdev** [comment]
So I have good and bad news.

The bad news is that I had hoped to implement this feature by the end of my holiday break. I didn't: my experiments with tooltips didn't work out and I had to go for custom drawing code like EclipseAceJump did; then I also found out that my initial plan to reuse Vrapper's completed motions is a no-go because they alter the editor's caret position while scanning for targets.

The good news is that I managed to build a proof of concept which at least gets some basics for the J&nbsp;/&nbsp;K&nbsp;/&nbsp;both motions in place:

![vrapper-easymotion-poc](https://user-images.githubusercontent.com/1451362/50566285-17cab100-0d38-11e9-9fe9-46c91d46f079.gif)

**Comment by albertdev** [comment]
@DXist is sadly right:
> @9527KGG , you may look into https://youtrack.jetbrains.com/issue/VIM-820 as well. Probably, JetBrains might be faster)

JetBrains seems to have coupled AceJump with IdeaVim using a new plugin: https://github.com/AlexPl292/IdeaVim-EasyMotion

Meanwhile I haven't really gotten around to finishing my work on the easymotion plugin, although it did seem to work better than what I reported last time around.

**Comment by 9527KGG** [comment]
@albertdev , I need Java and C++. Clion does not have a community version. So it does not work for me.

Is there any update BTW?

**Comment by tecfu** [comment]
Below is an example of how you can use [AceJump](https://github.com/tobiasmelcher/EclipseAceJump) to simulate EasyMotion's jumping to a matching character. 

In this example I map to the `f` key in normal mode:

```
"Emulate vim-easymotion using https://github.com/tobiasmelcher/EclipseAceJump
eclipseuiaction aceJumpChar ace.jump.CommandAnyChar
nmap f :aceJumpChar<enter>
```
